rcl/src/rcl/time.c

```diff
@@ -17,3 +17,54 @@
 #else  // defined(WIN32)
 #include "./time_unix.c"
 #endif  // defined(WIN32)
+
+#include "./stdatomic_helper.h"
+
+
+// Storage for ros time
+static atomic_bool __rcl_ros_time_is_active = ATOMIC_VAR_INIT(false);
+static atomic_uint_least64_t __rcl_ros_time_storage = ATOMIC_VAR_INIT(0);
+
+rcl_ret_t
+rcl_ros_time_point_now(rcl_ros_time_point_t * now)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(now, RCL_RET_INVALID_ARGUMENT);
+  if (!rcl_atomic_load_bool(&__rcl_ros_time_is_active)) {
+    rcl_system_time_point_t sys_time;
+    rcl_ret_t retcode = rcl_system_time_point_now(&sys_time);
+    now->nanoseconds = sys_time.nanoseconds;
+    return retcode;
+  }
+  now->nanoseconds = rcl_atomic_load_uint64_t(&__rcl_ros_time_storage);
+  return RCL_RET_OK;
+}
+
+rcl_ret_t
+rcl_set_ros_time_point_now(rcl_ros_time_point_t * now)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(now, RCL_RET_INVALID_ARGUMENT);
+  atomic_store(&__rcl_ros_time_storage, now->nanoseconds);
```

Comment:
You should probably use the `rcl_atomic_store` macro defined in `stdatomic_helper.h`. (applies in other places)